README.md
# http_code_util

Http code util is a simple dart plugin that abstracts http response status code checking for your projects



CHANGELOG.md
##[0.0.4] - Use mixin in place of class

## [0.0.3]

##[0.0.2] - Add checks for all status codes

## [0.0.1] - 14 / June / 2021.



test/http_code_util_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:http_code_util/http_code_util.dart';

class Test with StatusCodeUtil{
void main() {
  
test('verifies status code checks', () {
    expect(isSuccessful(200), true);
    expect(isSuccessful(300), false);

    expect(isClientError(456), true);
    expect(isClientError(399), false);

    expect(isInformation(102), true);
    expect(isInformation(99), false);

    expect(isServerError(512), true);
    expect(isServerError(443), false);

    expect(isRedirect(302), true);
    expect(isRedirect(202), false);

  });
}
}

lib/http_code_util.dart
library http_code_util;

import 'package:http_code_util/status_codes.dart';

/// @author: Ibekason Alexander Onyebuchi


mixin StatusCodeUtil {
  /// Returns true if code ranges between 200 and 300
  bool isSuccessful(int code) {
    return code >= 200 && code <= 299;
  }

  /// Returns true if code ranges between 100 and 200
  bool  isInformation(int code) {
    return code >= 100 && code <= 199;
  }

  /// Returns true if code ranges between 300 and 400
  bool isRedirect(int code) {
    return code >= 300 && code <= 399;
  }

  /// Returns true if code ranges between 400 and 499
  bool isClientError(int code) {
    return code >= 400 && code <= 499;
  }

  /// Returns true if code ranges between 500 and 599
  bool isServerError(int code) {
    return code >= 500 && code <= 599;
  }

  // information responses

  bool isContinue(int code) {
    return code == CONTINUE;
  }

  bool isSwitchProtocol(int code) {
    return code == SWITCHING_PROTOCOL;
  }

  bool isProcessing(int code) {
    return code == PROCESSING;
  }

  bool isEarlyHints(int code) {
    return code == EARLY_HINTS;
  }

  // successful responses

  bool isOk(int code) {
    return code == OK;
  }

  bool isCreated(int code) {
    return code == CREATED;
  }

  bool isAccepted(int code) {
    return code == ACCEPTED;
  }

  bool isNonAuthInfo(int code) {
    return code == NONAUTHORITATIVE_INFORMATION;
  }

  bool isNoContent(int code) {
    return code == NO_CONTENT;
  }

  bool isResetContent(int code) {
    return code == RESET_CONTENT;
  }

  bool isPartialContent(int code) {
    return code == PARTIAL_CONTENT;
  }

  bool isMultiStatus(int code) {
    return code == MULTI_STATUS;
  }

  bool isAlreadyReported(int code) {
    return code == ALREADY_REPORTED;
  }

  bool isImUsed(int code) {
    return code == IM_USED;
  }

  // Redirection Responses

  bool isMultiChoice(int code) {
    return code == MULTIPLE_CHOICE;
  }

  bool isMoved(int code) {
    return code == MOVED_PERMANENTLY;
  }

  bool isFound(int code) {
    return code == FOUND;
  }

  bool isSeeOther(int code) {
    return code == SEE_OTHER;
  }

  bool isNotModified(int code) {
    return code == NOT_MODIFIED;
  }

  bool isUseProxy(int code) {
    return code == USE_PROXY;
  }

  bool isUnused(int code) {
    return code == UNUSED;
  }

  bool isTempRedirect(int code) {
    return code == TEMPORARY_REDIRECT;
  }

  bool isPermRedirect(int code) {
    return code == PERMANENT_REDIRECT;
  }

  // Client error responses

  bool isBadRequest(int code) {
    return code == BAD_REQUEST;
  }

  bool isForbidden(int code) {
    return code == FORBIDDEN;
  }

  bool isUnauthorized(int code) {
    return code == UNAUTHORIZED;
  }

  bool isPaymentRequired(int code) {
    return code == PAYMENT_REQUIRED;
  }

  bool isNotFound(int code) {
    return code == NOT_FOUND;
  }

  bool isMethodNotAllowed(int code) {
    return code == METHOD_NOTALLOWED;
  }

  bool isNotAcceptable(int code) {
    return code == NOT_ACCEPTABLE;
  }

  bool isProxyAuthRequired(int code) {
    return code == PROXY_AUTH_REQUIRED;
  }

  bool isRequestTimeout(int code) {
    return code == REQUEST_TIMEOUT;
  }

  bool isConflict(int code) {
    return code == CONFLICT;
  }

  bool isGone(int code) {
    return code == GONE;
  }

  bool isLengthRequired(int code) {
    return code == LENGTH_REQUIRED;
  }

  bool isPreconditionedFailed(int code) {
    return code == PRECONDITION_FAILED;
  }

  bool isPayloadTooLarge(int code) {
    return code == PAYLOAD_TOO_LARGE;
  }

  bool isUriTooLong(int code) {
    return code == URI_TOO_LONG;
  }

  bool isUnsupportedMedia(int code) {
    return code == UNSUPPORTED_MEDIA_TYPE;
  }

  bool isRangeNotSatisfiable(int code) {
    return code == RANGE_NOT_SATISFIABLE;
  }

  bool isExpectationFailed(int code) {
    return code == EXPECTATION_FAILED;
  }

  bool isImATeapot(int code) {
    return code == IMATEAPOT;
  }

  bool isMisdirectedRequest(int code) {
    return code == MISDIRECTED_REQUEST;
  }

  bool isUnprocessedEntity(int code) {
    return code == UNPROCESSABLE_ENTITY;
  }

  bool isLocked(int code) {
    return code == LOCKED;
  }

  bool isFailedDependency(int code) {
    return code == FAILED_DEPENDENCY;
  }

  bool isTooEarly(int code) {
    return code == TOO_EARLY;
  }

  bool isUpgradeRequired(int code) {
    return code == UPGRADE_REQUIRED;
  }

  bool isPreconditionRequired(int code) {
    return code == PRECONDITION_REQUIRED;
  }

  bool isTooManyRequests(int code) {
    return code == TOO_MANY_REQUESTS;
  }

  bool isRequestHeaderLong(int code) {
    return code == REQUEST_HEADER_FILES_TOO_LONG;
  }

  bool isLegalReasons(int code) {
    return code == LEGAL_REASONS;
  }

  // server responses

  bool isInternalServerError(int code) {
    return code == INTERNAL_SERVER_ERROR;
  }

  bool isNotImplemented(int code) {
    return code == NOT_IMPLEMENTED;
  }

  bool isBadGateway(int code) {
    return code == BAD_GATEWAY;
  }

  bool isServiceUnavailable(int code) {
    return code == SERVICE_UNAVAILABLE;
  }

  bool isGatewayTimeout(int code) {
    return code == GATEWAY_TIMEOUT;
  }

  bool isHttpNotSupported(int code) {
    return code == HTTP_NOT_SUPPORTED;
  }

  bool isVariantAlso(int code) {
    return code == VARIANT_ALSO_NEGOTIATES;
  }

  bool isSufficientStorage(int code) {
    return code == INSUFFICIENT_STORAGE;
  }

  bool isLoopDetected(int code) {
    return code == LOOP_DETECTED;
  }

  bool isNotExtended(int code) {
    return code == NOT_EXTENDED;
  }

  bool isNetworkAuthError(int code) {
    return code == NETWORK_AUTHENTICATION_ERROR;
  }
}


lib/status_codes.dart
//information responses

const int CONTINUE = 100;
const int SWITCHING_PROTOCOL = 101;
const int PROCESSING = 102;
const int EARLY_HINTS = 103;

// Successful responses
const int OK = 200;
const int CREATED = 201;
const int ACCEPTED = 202;
const int NONAUTHORITATIVE_INFORMATION = 203;
const int NO_CONTENT = 204;
const int RESET_CONTENT = 205;
const int PARTIAL_CONTENT = 206;
const int MULTI_STATUS = 207;
const int ALREADY_REPORTED = 208;
const int IM_USED = 226;

// Redirection responses
const int MULTIPLE_CHOICE = 300;
const int MOVED_PERMANENTLY = 301;
const int FOUND = 302;
const int SEE_OTHER = 303;
const int NOT_MODIFIED = 304;
const int USE_PROXY = 305;
const int UNUSED = 306;
const int TEMPORARY_REDIRECT = 307;
const int PERMANENT_REDIRECT = 308;

// Client error responses
const int BAD_REQUEST = 400;
const int UNAUTHORIZED = 401;

const int PAYMENT_REQUIRED = 402;
const int FORBIDDEN = 403;
const int NOT_FOUND = 404;
const int METHOD_NOTALLOWED = 405;
const int NOT_ACCEPTABLE = 406;
const int PROXY_AUTH_REQUIRED = 407;
const int REQUEST_TIMEOUT = 408;
const int CONFLICT = 409;
const int GONE = 410;
const int LENGTH_REQUIRED = 411;
const int PRECONDITION_FAILED = 412;
const int PAYLOAD_TOO_LARGE = 413;
const int URI_TOO_LONG = 414;
const int UNSUPPORTED_MEDIA_TYPE = 415;
const int RANGE_NOT_SATISFIABLE = 416;
const int EXPECTATION_FAILED = 417;
const int IMATEAPOT = 418;
const int MISDIRECTED_REQUEST = 421;
const int UNPROCESSABLE_ENTITY = 422;
const int LOCKED = 423;
const int FAILED_DEPENDENCY = 424;
const int TOO_EARLY = 425;
const int UPGRADE_REQUIRED = 426;
const int PRECONDITION_REQUIRED = 428;
const int TOO_MANY_REQUESTS = 429;
const int REQUEST_HEADER_FILES_TOO_LONG = 431;
const int LEGAL_REASONS = 451;

// server responses
const int INTERNAL_SERVER_ERROR = 500;
const int NOT_IMPLEMENTED = 501;

const int BAD_GATEWAY = 502;

const int SERVICE_UNAVAILABLE = 503;

const int GATEWAY_TIMEOUT = 504;

const int HTTP_NOT_SUPPORTED = 505;

const int VARIANT_ALSO_NEGOTIATES = 506;

const int INSUFFICIENT_STORAGE = 507;

const int LOOP_DETECTED = 508;

const int NOT_EXTENDED = 510;
const int NETWORK_AUTHENTICATION_ERROR = 511;
